.github/workflows/e2e.yaml

```
@@ -101,6 +102,9 @@ jobs:
           k8s_version: ${{ inputs.k8s_version }}
           ip_family: ${{ inputs.suite == 'IPv6' && 'IPv6' || 'IPv4' }} # Set the value to IPv6 if IPv6 suite, else IPv4
           git_ref: ${{ inputs.git_ref }}
+      - name: cloudformation describe stack events
+        if: ${{ failure() && steps.create-eks-cluster.outcome != 'success' }}
```

```suggestion
        if: ${{ steps.create-eks-cluster.outcome != 'success' }}
```

Do we need both of these checks or can we just rely on the second one?

---

We would need both as just the second one skips the steps because of a default `sucess()` check being applied.
[Docs state that](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions): 

> A default status check of success() is applied unless you include one of these functions.

---

Doesn't that mean that if you don't have an `if` condition, that it would use `success()`? I think in this case, since we have an `if` statement, we can just assume that we can check a single step to see if it failed and base whether we run this step on that

---

Talked offline on Slack. Github has a section that details that both of these conditions need to be there: https://docs.github.com/en/actions/learn-github-actions/expressions#failure-with-conditions